### PR TITLE
Update the default value of `unsafeDisableTooltip` to `false` and remove redundant props and wrappers

### PR DESCRIPTION
**Pull request by broccolinisoup**
Closes https://github.com/primer/react/issues/2008
Related https://github.com/primer/react/issues/4696

> [!NOTE]
> We managed to complete the roll out at dotcom  🎉  (View the progress [here (internal link only)](https://github.com/github/primer/discussions/3333))

#### Changed

- The default value of `unsafeDisableTooltip` to false
- Added `unsafeDisableTooltip={true}` a coupe of IconButton instances where it is not safe to enable tooltips.

#### Removed

- Explicit Tooltip wrappers
- `unsafeDisableTooltip={false}` prop since it is the default value now.

### Rollout strategy

- [ ] Patch release
- [x] Minor release
- [ ] Major release; if selected, include a written rollout or migration plan
- [ ] None; if selected, include a brief description as to why

### Merge checklist

- [x] Added/updated tests
- [ ] Added/updated documentation
- [x] Added/updated previews (Storybook)
- [ ] Changes are [SSR compatible](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility)
- [x] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge
- [ ] (GitHub staff only) Integration tests pass at github/github ([Learn more about how to run integration tests](https://github.com/github/primer-engineering/blob/main/how-we-work/testing-primer-react-pr-at-dotcom.md))

### CODE REVIEW

File: packages/react/src/__tests__/Dialog.test.tsx [hunk 1]
```diff
@@ -140,12 +140,24 @@ describe('Dialog', () => {
...
+    const tooltipEl = getByText('Close')
+    // first ESC closes the tooltip
+    const user = userEvent.setup()
+    await user.keyboard('{Escape}')
+    await waitFor(() => {
+      // Wait for the tooltip to close
+      expect(tooltipEl).not.toHaveAttribute(':popover-open')
+    })
+
+    // second ESC closes the dialog and returns focus
```
Comment (broccolinisoup) [inline packages/react/src/__tests__/Dialog.test.tsx:155]: This test is [failing](https://github.com/primer/react/actions/runs/9707822928/job/26793694928?pr=4702#step:7:819) - Not sure how to address this. I am trying to await before pressing the second ESC but it doesn't seem to be working. Any suggestions, thoughts? 🤔

File: packages/react/src/__tests__/Dialog.test.tsx [hunk 2]
```diff
@@ -140,7 +169,9 @@ describe('Dialog', () => {
   })
 
   it('Returns focus to returnFocusRef on escape', async () => {
-    const {getByTestId, queryByTestId} = HTMLRender(<Component />)
```
Comment (broccolinisoup) [inline packages/react/src/__tests__/Dialog.test.tsx:143]: I had an issue with this existing component. It was supposed to go back to the return ref after pressing the ESC twice (First closing the tooltip, second closing the dialog) but for some reason it didn't work, even though I waited for the previous keyboard event to be completed. This test wasn't about the tooltip itself; it is about the Dialog so I updated to use a different dialog but let me know if I am misinterpreting the issue I encountered.

File: packages/react/src/Dialog.tsx [hunk 3]
```diff
@@ -11,6 +11,7 @@ import type {ComponentProps} from './utils/types'
 import {useRefObjectAsForwardedRef} from './hooks/useRefObjectAsForwardedRef'
 import {XIcon} from '@primer/octicons-react'
 
+// Dialog v1
```
Comment (siddharthkp) [inline packages/react/src/Dialog.tsx:14]: Classic! 😅

**Review by siddharthkp — APPROVED** [review]
Looks good! Does what it says on the box, good work!

Lets run some integration tests with this before we merge

**Comment by broccolinisoup** [comment]
Integration tests pass https://github.com/github/github/pull/332110 and tested on RL, everything looks good 🎉

**Pull request merged**